edu_edfi_airflow/dags/earthbeam_dag.py

```diff
@@ -78,6 +79,81 @@ def build_local_raw_dir(self, tenant_code: str, api_year: int, grain_update: Opt
         )
         return raw_dir
 
+    @staticmethod
+    def partition_on_tenant_and_year(
+        csv_paths: Union[Union[str, Path], List[Union[str, Path]]],
+        output_dir: str,
+        tenant_col: str = "tenant_code",
+        tenant_map: dict = None,
+        year_col: str = "api_year",
+        year_map: dict = None,
+    ):
+        """
+        Preprocessing function to shard data to parquet on disk
+
+        :param csv_paths: one or more complete file paths pointing to input data
+        :param output_dir: root directory of the parquet
+        :param tenant_col: (optional) name of the column to use as tenant code
+        :param tenant_map: (optional) map values from the contents of tenant_col to valid tenant codes
+        :param year_col: (optional) name of the column to use as API year
+        :param year_map: (optional) map values from the contents of api_col to valid API years
+
+        :return:
+        """
+        # (expensive) imports here so that the airflow scheduler doesn't have to deal with them
+        import dask
+        import dask.dataframe as dd
+        import pandas as pd
+
+        tenant_code = "tenant_code"
+        api_year = "api_year"
+
+        input_is_list = True if isinstance(csv_paths, list) else False
```

Comment:
Do we need this boolean or conditional? Can we not just use the following to conform the variable to a list?
```
csv_paths = list(csv_paths)
```

Reply:
In the case of `csv_paths` being a string, wouldn't that split it into a list of single characters? e.g. `['d', 'a', 't', 'a', '.', 'c', 's', 'v']`?

Reply:
Yep, you're correct here. I had hoped that that was not the case, but Python makes me a fool once again.

I'm trying to avoid an additional variable defined if we can just force the datatype of the input to be consistent. Maybe alternatively this could be done:
```
csv_paths = [csv_paths] if isinstance(csv_paths, str) else csv_paths
```

Reply:
Did something slightly different but still a one-liner; good thinking!